js/views/reader_view.js

```diff
@@ -177,6 +177,8 @@ ReadiumSDK.Views.ReaderView = function(options) {
 
             if (!ReadiumSDK.Helpers.isIframeAlive($iframe[0])) return;
 
+            self.trigger(ReadiumSDK.Events.CONTENT_DOCUMENT_LOADED_BEFORE_INJECTION, $iframe, spineItem);
```

Comment:
Note that in the `feature/plugins` branch, the `CONTENT_DOCUMENT_LOADED` event is used to `attachAnnotations()` in the new "annotations" plugins:
https://github.com/readium/readium-shared-js/blob/feature/pluginsX/plugins/annotations/main.js#L104
This replaces `_annotationsManager.attachAnnotations($iframe, spineItem);` a few lines below (see
https://github.com/readium/readium-shared-js/pull/98/files#diff-7f14747656265ce3e39fee34e732d161R186 )

So, although `CONTENT_DOCUMENT_LOADED_BEFORE_INJECTION` is a nice addition, it is not strictly necessary for plugins like "popup footnotes" to work. Hooking into `CONTENT_DOCUMENT_LOADED` is fine in most cases, it just means that at this point in time, injected behaviours like Media Overlays, link processor, triggers, switches, etc. have already been applied.
Also note that there is not guarantee that (for example) annotations behaviour is injected before or after popup footnotes (etc.), so plugins cannot rely on such ordering assumptions.